test/suites/nodeclaim/suite_test.go

```diff
@@ -0,0 +1,51 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodeclaim_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/Azure/karpenter/pkg/apis/v1alpha2"
+	"github.com/Azure/karpenter/test/pkg/environment/azure"
+	corev1beta1 "github.com/aws/karpenter-core/pkg/apis/v1beta1"
+)
+
+var env *azure.Environment
+var nodeClass *v1alpha2.AKSNodeClass
+var nodePool *corev1beta1.NodePool
+
+func TestNodeClaim(t *testing.T) {
+	RegisterFailHandler(Fail)
+	BeforeSuite(func() {
+		env = azure.NewEnvironment(t)
+	})
+	AfterSuite(func() {
+		env.Stop()
```

Comment:
Again not for this PR, but the Integration E2E test suite is the only other one that has `env.Stop()`. Should look at if this should be minored to the others. Looking at the code its a context cancel, so probably good to have regardless at the end of each suite:
https://github.com/Azure/karpenter/blob/19d06fd74448bcdee193a693572c4043780329e4/test/pkg/environment/common/environment.go#L92C9-L92C9

Reply:
agreed